express-notes-api/index.js

```diff
@@ -0,0 +1,142 @@
+const express = require('express');
+const app = express();
+
+const data = require('./data.json');
+var nextID = data.nextId;
+var notes = data.notes;
+
+app.get('/api/notes', (req, res) => {
+  const dataArray = [];
+  for (const prop in notes) {
+    dataArray.push(notes[prop]);
+  }
+  res.json(dataArray);
+});
+
+app.get('/api/notes/:id', (req, res) => {
+  const id = req.params.id;
+  const isInt = Number.isInteger(parseInt(id));
+  let idIsThere = false;
+  var noteToRead;
+
+  const errPostiveInt = {
+    error: 'id must be a positive integer'
+  };
+  const errCannotFind = {
+    error: `cannot find note with id ${id}`
+  };
+
+  for (const prop in notes) {
+    if (parseInt(prop) === parseInt(id)) {
+      idIsThere = true;
+    }
+  }
+
+  if ((isInt === false) || parseInt(id) < 0) {
+    res.status(400).json(errPostiveInt);
```

Comment:
Requests like `GET /api/notes/2.789` should still result in a `400 Bad Request` response.